### PR TITLE
Added filtering to appleseed.studio unit tests (Issue #1359)

### PR DESCRIPTION
**Pull request by htmlboss**
Primary changes:
- Unit tests in appleseed.studio are filtered by regex in realtime (new feature).
- const-corrected TestWindow class.
- Removed unused header files in TestWindow.h/.cpp.
- Various minor fixes to code layout to keep things consistent.

Screenshot of relevant UI changes (I tried to follow the same layout from the main window):

ui

### CODE REVIEW

File: src/appleseed.studio/debug/tests/testwindow.cpp [hunk 1]
```diff
@@ -365,19 +374,56 @@ namespace
...
+{
+    bool do_filter_items(QTreeWidgetItem* item, const QRegExp& regexp)
+    {
+        bool any_children_visible = false;
+
+        for (int i = 0; i < item->childCount(); ++i)
+        {
+            if (do_filter_items(item->child(i), regexp))
+            {
+                any_children_visible = true;
```
Comment (dictoon) [inline src/appleseed.studio/debug/tests/testwindow.cpp]: You can add a `break` here to exit the loop as soon as one visible child is found.

File: src/appleseed.studio/debug/tests/testwindow.cpp [hunk 2]
```diff
@@ -307,7 +314,9 @@ namespace
         {
             QTreeWidgetItem* item = parent->child(i);
             if (item->checkState(column) == state)
+            {
```
Comment (dictoon) [inline src/appleseed.studio/debug/tests/testwindow.cpp]: We're actually fine with omitting the curly braces for single line blocks.

**Comment by dictoon** [comment]
Thanks for the super nice PR! The code is extremely clear and I have literally nothing to complain about in this area.

A few remarks on the UI/UX front:

* For the sake of consistency with the project explorer and the color/texture picker (two places where filtering is available), I would suggest placing the textbox and button _above_ the unit test list rather than below.

* Another thing to consider is whether we should add a way to check/uncheck the tests that are shown in the list _after filtering_. What do you think?

**Comment by htmlboss** [comment]
> Another thing to consider is whether we should add a way to check/uncheck the tests that are shown in the list after filtering. What do you think?

In terms of convenience I really like that. Are you suggesting adding buttons to select/de-select all items presently shown in the list (as opposed to toggling all loaded tests)? It would be very easy to implement, but wouldn't that clutter the UI though?

**Comment by dictoon** [comment]
Maybe we should actually modify the existing buttons to operate on the shown tests instead of the whole test suite?

**Comment by dictoon** [comment]
Merged, thank you!

**Pull request merged**